BlueprintUI/Sources/Layout/Aligned.swift

```diff
@@ -113,9 +113,71 @@ public struct Aligned: Element {
 
             return attributes
         }
+
+        func sizeThatFits(proposal: SizeConstraint, subelement: LayoutSubelement, cache: inout Cache) -> CGSize {
+            subelement.sizeThatFits(proposal)
+        }
+
+        func placeSubelement(in size: CGSize, subelement: LayoutSubelement, cache: inout ()) {
+            let x: CGFloat
+            let y: CGFloat
+
+            let subelementSize = subelement.sizeThatFits(SizeConstraint(size))
+            let clampedSize = subelementSize.upperBounded(maxWidth: size.width, maxHeight: size.height)
+
+            let width: CGFloat
+            let height: CGFloat
+
+            switch horizontalAlignment {
+            case .leading:
+                x = 0
+                width = clampedSize.width
+            case .center:
+                x = 0.5
+                width = clampedSize.width
+            case .trailing:
+                x = 1
+                width = clampedSize.width
+            case .fill:
+                x = 0
+                width = size.width
+            }
+
+            switch verticalAlignment {
+            case .top:
+                y = 0
+                height = clampedSize.height
+            case .center:
+                y = 0.5
+                height = clampedSize.height
+            case .bottom:
+                y = 1
+                height = clampedSize.height
+            case .fill:
+                y = 0
+                height = size.height
+            }
+
+            let position = CGPoint(x: x * size.width, y: y * size.height)
```

Comment:
For my own education / confirmation: This works because of the `UnitPoint`, right? Eg I am thinking through:

- I have an element of width 50
- It's inside an `Aligned` of width 200
- I align `.center`
- `position.x` will then be `25`

I'd actually expect this to be `(200 - 50) / 2 == 75`

But the `UnitPoint` ends up forcing the right positioning?

Reply:
I think `position.x` is 100 in your example. Since `x` is 0.5, the horizontal center of the subelement will be positioned at x: 100 (the middle of the `Aligned`), which is correct. At least that's how I understand it.